pkg/apis/triggers/v1alpha1/event_listener_validation.go

```diff
@@ -59,11 +67,82 @@ func validateKubernetesObject(orig *KubernetesResource) (errs *apis.FieldError)
 	if len(orig.Template.Spec.Containers) == 1 {
 		errs = errs.Also(apis.CheckDisallowedFields(orig.Template.Spec.Containers[0],
 			*containerFieldMask(&orig.Template.Spec.Containers[0])).ViaField("spec.template.spec.containers[0]"))
+		// validate env
+		errs = errs.Also(validateEnv(orig.Template.Spec.Containers[0].Env).ViaField("spec.template.spec.containers[0].env"))
 	}
 
 	return errs
 }
 
+func validateEnv(envVars []corev1.EnvVar) (errs *apis.FieldError) {
+	var (
+		count    = 0
+		envValue string
+	)
+	for i, env := range envVars {
+		errs = errs.Also(validateEnvVar(env).ViaIndex(i))
+		if reservedEnvVars.Has(env.Name) {
+			count++
+			envValue = env.Name
+		}
+	}
+	// This is to make sure both TLS_CERT and TLS_KEY is set for tls connection
+	if count == 1 {
+		errs = errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("Expected env's are TLS_CERT and TLS_KEY, but got only one env %s", envValue),
+		})
+	}
+	return errs
+}
+
+func validateEnvVar(env corev1.EnvVar) (errs *apis.FieldError) {
+	errs = errs.Also(apis.CheckDisallowedFields(env, *envVarMask(&env)))
+
+	return errs.Also(validateEnvValueFrom(env.ValueFrom).ViaField("valueFrom"))
+}
+
+func validateEnvValueFrom(source *corev1.EnvVarSource) *apis.FieldError {
+	if source == nil {
+		return nil
+	}
+	return apis.CheckDisallowedFields(*source, *envVarSourceMask(source))
+}
+
+// envVarSourceMask performs a _shallow_ copy of the Kubernetes EnvVarSource object to a new
+// Kubernetes EnvVarSource object bringing over only the fields allowed in the Triggers EventListener API.
+func envVarSourceMask(in *corev1.EnvVarSource) *corev1.EnvVarSource {
+	if in == nil {
+		return nil
+	}
+	out := new(corev1.EnvVarSource)
+	// Allowed fields
+	out.SecretKeyRef = in.SecretKeyRef
+
+	// Disallowed fields
+	out.ConfigMapKeyRef = nil
+	out.FieldRef = nil
+	out.ResourceFieldRef = nil
+
+	return out
+}
```

Comment:
I own to refactor changes related to masking fields in a follow up PR by creating separate file so that `_validation.go` will have changes related to validation